lti_consumer/lti_1p3/consumer.py

```diff
@@ -659,3 +661,228 @@ def enable_nrps(self, context_memberships_url):
+        claim_key = "https://purl.imsglobal.org/spec/lti/claim/version"
+        check_token_claim(
+            proctoring_response,
+            claim_key,
+            "1.3.0",
+            f"Token's {claim_key} claim should be 1.3.0."
+        )
+
+        # Check that the response session_data claim is the correct anti-CSRF token.
+        claim_key = "https://purl.imsglobal.org/spec/lti-ap/claim/session_data"
+        check_token_claim(
+            proctoring_response,
+            claim_key,
+            self.proctoring_data.get("session_data"),
+            f"Token's {claim_key} claim is not correct."
+        )
+
+        # TODO: Right now, the library doesn't support additional claims within the resource_link claim.
+        #       Once it does, we should check the entire claim instead of just the id.
+        claim_key = "https://purl.imsglobal.org/spec/lti/claim/resource_link"
+        check_token_claim(
+            proctoring_response,
+            claim_key,
+            {"id": self.proctoring_data.get("resource_link")},
+            f"Token's {claim_key} claim is not correct."
+        )
+
+        claim_key = "https://purl.imsglobal.org/spec/lti-ap/claim/attempt_number"
+        check_token_claim(
+            proctoring_response,
+            claim_key,
+            self.proctoring_data.get("attempt_number"),
+            f"Token's {claim_key} claim is not correct."
+        )
+
+        # -------------------------
+        # Check Optional LTI Claims
+        # -------------------------
+
+        verified_user = proctoring_response.get("https://purl.imsglobal.org/spec/lti-ap/claim/verified_user", {})
+        # See 4.3.2.1 Verified user claim.
+        # The iss and sub attributes SHOULD NOT be included as they are opaque to the Proctoring Tool and cannot be
+        # independently verified.
+        iss = verified_user.get('iss')
+        if iss is not None:
+            raise exceptions.InvalidClaimValue('Token verified_user claim should not contain the iss claim.')
+        sub = verified_user.get('sub')
+        if sub is not None:
+            raise exceptions.InvalidClaimValue('Token verified_user claim should not contain the sub claim.')
+
+        # See 4.3.2.1 Verified user claim.
+        # If the picture attribute is provided it MUST point to a picture taken by the Proctoring Tool.
+        # It MUST NOT be the same picture provided by the Assessment Platform in the Start Proctoring message.
+        picture = verified_user.get('picture')
+        if picture and picture == self.lti_claim_user_data.get('picture'):
+            raise exceptions.InvalidClaimValue(
+                'If the verified_claim is provided and contains the picture claim,'
+                ' the picture claim should not be the same picture provided by the Assessment Platform to the Tool.'
+            )
+        # TODO: We can leverage these verified user claims. For example, we could use
+        #       these claims for Name Affirmation.
```

Comment:
it might make sense to pull out the claim check in to a separate function which would be easier to test by itself

it could return the verified user we could just get that out again

Reply:
Are you referring to lines 856 - 877?

Reply:
yep!